### PR TITLE
Gpt oss

### LINKED ISSUE #143: GPT-OSS
## Model description
 GPT-OSS

## Open source examples / papers / references
* Paper: { https://arxiv.org/pdf/2508.10925 }
* Website: { https://openai.com/index/introducing-gpt-oss/ }
* Model code: { https://github.com/jax-ml/jax-llm-examples/tree/main/gpt_oss }
* Model weights: { https://huggingface.co/openai/gpt-oss-120b }

## Why this model?
* GPT-OSS-120B is the first fully open-weight, OpenAI-released frontier-scale language model

### PR DESCRIPTION
**Pull request by Aatman09**
Resolves #143 

**Reference**

I have implemented the modeling.py and the params.py code and yet to commit the test_outputs.py and run_model.py

**Checklist**

- [X] I have read the **[Contribution Guidelines](https://github.com/jax-ml/bonsai/blob/main/CONTRIBUTING.md#contributing-a-model)** and used [pre-commit hooks](https://github.com/jax-ml/bonsai/blob/main/CONTRIBUTING.md#linting-and-type-checking) to format this commit.
- [ ] I have added all the necessary **unit tests** for my change. (`run_model.py` for model usage, `test_outputs.py` and/or `model_validation_colab.ipynb` for quality).
- [X] **(If using an LLM)** I have carefully reviewed and removed all **superfluous comments** or unneeded, commented-out code. Only necessary and functional code remains.
- [X] I have signed the **[Contributor License Agreement (CLA)](https://cla.developers.google.com/about)**.

### CODE REVIEW

**Comment by coder0143** [comment]
@Aatman09 Good to have GPT-OSS here, do add tests for - layers, attention, and final logits and a run_model.py file. Since this is an MoE, do add sharding if possible.
